datafusion/datasource-parquet/src/file_format.rs

```diff
@@ -350,15 +352,17 @@ impl FileFormat for ParquetFormat {
             Some(time_unit) => Some(parse_coerce_int96_string(time_unit.as_str())?),
             None => None,
         };
-        let config_file_decryption_properties = &self.options.crypto.file_decryption;
+        #[cfg(feature = "parquet_encryption")]
         let file_decryption_properties: Option<FileDecryptionProperties> =
-            match config_file_decryption_properties {
+            match &self.options.crypto.file_decryption {
                 Some(cfd) => {
                     let fd: FileDecryptionProperties = cfd.clone().into();
                     Some(fd)
                 }
                 None => None,
             };
+        #[cfg(not(feature = "parquet_encryption"))]
+        let file_decryption_properties: Option<FileDecryptionProperties> = None;
```

Comment:
Does this build when encryption is disabled? If encryption is disabled I would expect that the `FileDecryptionProperties` type wouldn't be defined so this shouldn't build.

And `fetch_schema_with_location` below shouldn't build either as it shouldn't know about the `FileDecryptionProperties` type when encryption is disabled.

Reply:
Not true. If the option is disabled then FileDecryptionProperties is just defined to be empty.

Reply:
Ah sorry I didn't see that, I should have held off commenting before reading the whole diff!